index.html

```diff
@@ -3012,11 +3099,11 @@ <h3>Receiving and parsing content</h3>
               </li>
               <li>
                 If <var>charset</var> is equal to <code>"utf-8"</code>,
-                set <var>record</var>'s data to the result of running
-                <a>UTF-8 decode</a> on <var>buffer</var>.
+                set <var>record</var>'s associated <a>bytes</a> to <var>buffer</var>.
               </li>
               <li>
-                Otherwise, set <var>record</var>'s data to the result of running
+                Otherwise, set the result of invoking <var>record</var>'s
+                <a href="#dom-ndefrecord-totext">toText()</a> method to the result of running
```

Comment:
Yeah this part is really weird. You are replacing how calls to that future in the future will work. Much better to just set bytes to buffer in all cases (would simplify this algorithm) and then in the toText() method, check the mimetype for the charset and then choose the right decoding method